### ISSUE TITLE
Padding, strides and dilation rates

### ISSUE THREAD

**Issue by FrederikDegn** [lich99/Stock_CNN#2]
Hi 

I was curious as to how you ended up with a padding of (12,1) in the baseline model. I get that you'd have to sustain that padding across the blocks with your setup, but in the paper they only use stride and dilation in the first layer and not the subsequent ones. 

And do you know which paddings to use for 5d and 60d horizons?

**Comment by lich99** [comment]
Hi FrederikDegn, 

I think you're right. For,
> _We use vertical strides of 1, 3, and 3, and vertical dilation rates of 1, 2,
> and 3 for 5-day, 20-day, and 60-day images, respectively, only on the first layer, where inputs
> are sparse raw images._

To maintain the shape of the matrix, (8,1), (2,1), and (2,1) paddings should work in the 20d horizon.
Unfortunately, I don't find the specific settings in the paper as well, just guess.

**Comment by FrederikDegn** [comment]
Hi again

The specific setting in their paper is in appendix footnote 17, p. 44:

> For elements at the image’s border, we fill in the absent neighbor elements with zeros in order to compute the convolution. This is a common CNN practice known as “padding,” and ensures that the convolution output has the same dimension as the image itself (see Simonyan and Zisserman, 2015).

This is somehow easier to implement in tensorflow...

As I went for replicating the whole paper incl. creating images from CRSP, and thus also looking at 5d and 60d horizon, I opted for a solution that automatically works around the problem of pytorch not having padding="same" as tensorflow. An answer to my issue [here](https://stackoverflow.com/questions/74029235/unexpected-number-of-fully-connected-neurons-after-padding-with-stride-in-pytorc) solved my problem of padding dimensions across horizons. 

Thanks for answering, I'm closing the issue.
